src/plugin.js

```diff
@@ -0,0 +1,317 @@
+ *
+ * @param  {Number} n
+ * @return {Boolean}
+ */
+const isInteger = n => n >= 0 && n % 1 === 0;
+
+/**
+ * Whether a value is a string with no whitespace.
+ *
+ * @param  {String} s
+ * @return {Boolean}
+ */
+const hasNoWhitespace = s => typeof s === 'string' && (/^\S+$/).test(s);
+
+/**
+ * Overlay component.
+ *
+ * @class   Overlay
+ * @extends {videojs.Component}
+ */
+class Overlay extends Component {
+
+  constructor(player, options) {
+    super(player, options);
+
+    player.addChild(this);
+
+    ['start', 'end'].forEach(key => {
+      let value = this.options_[key];
+
+      if (isInteger(value)) {
+        this[key + 'Event_'] = 'timeupdate';
+      } else if (hasNoWhitespace(value)) {
+        this[key + 'Event_'] = value;
+
+      // An overlay MUST have a start option. Otherwise, it's pointless.
+      } else if (key === 'start') {
+        throw new Error('invalid "start" option; expected number or string');
+      }
+    });
+
+    // video.js does not like components with multiple instances binding
+    // events to the player because it tracks them at the player level,
+    // not at the level of the object doing the binding. This could also be
+    // solved with Function.prototype.bind (but not videojs.bind because of
+    // its GUID magic), but the anonymous function approach avoids any issues
+    // caused by crappy libraries clobbering Function.prototype.bind.
+    // - https://github.com/videojs/video.js/issues/3097
+    ['endListener_', 'rewindListener_', 'startListener_'].forEach(name => {
+      this[name] = (e) => Overlay.prototype[name].call(this, e);
+    });
+
+    // If the start event is a timeupdate, we need to watch for rewinds (i.e.,
+    // when the user seeks backward).
+    if (this.startEvent_ === 'timeupdate') {
+      this.on(player, 'timeupdate', this.rewindListener_);
+    }
+
+    this.debug(tsmlj`
+      created, listening to "${this.startEvent_}" for "start" and
+      "${this.endEvent_ || 'nothing'}" for "end"
+    `);
+
+    this.hide();
+  }
+
+  createEl() {
+    let options = this.options_;
+    let content = options.content;
+
+    let el = videojs.createEl('div', {
+      className: `vjs-overlay vjs-overlay-${options.align} ${options.class} vjs-hidden`
+    });
+
+    if (typeof content === 'string') {
+      el.innerHTML = content;
+    } else if (content instanceof DocumentFragment) {
+      el.appendChild(content);
+    } else {
+      videojs.appendContent(el, content);
+    }
+
+    return el;
+  }
+
+  /**
+   * Logs debug errors
+   * @param  {...[type]} args [description]
+   * @return {[type]}         [description]
+   */
+  debug(...args) {
+    if (!this.options_.debug) {
+      return;
+    }
+
+    let log = videojs.log;
+    let fn = log;
+
+    // Support `videojs.log.foo` calls.
+    if (log.hasOwnProperty(args[0]) && typeof log[args[0]] === 'function') {
+      fn = log[args.shift()];
+    }
+
+    args.unshift(`overlay#${this.id()}: `);
+    fn(...args);
```

Comment:
Ah, yes, good call. I'll include this change as well. Might as well, while I'm in there.